solver/progress.go

```diff
@@ -164,25 +167,29 @@ type syncProgressPrinter struct {
 
 var _ progress.Writer = (*syncProgressPrinter)(nil)
 
-func newSyncProgressPrinter(w io.Writer, out console.File, mode string) *syncProgressPrinter {
+func newSyncProgressPrinter(w io.Writer, out console.File, mode string) (*syncProgressPrinter, error) {
 	spp := &syncProgressPrinter{
 		w:    w,
 		out:  out,
 		mode: mode,
 	}
-	spp.reset()
-	return spp
+	if err := spp.reset(); err != nil {
+		return nil, err
+	}
+	return spp, nil
```

Comment:
Can this just be `return spp, spp.reset()`?

Reply:
Done